app/services/create_post.rb

```diff
@@ -1,6 +1,6 @@
 class CreatePost
   def self.call(group, membership, post_type_id)
-    if post_type_id == PostType::LEADER_POST_ID
+    if post_type_id == PostType::LEADER_POST_ID && group.leader.present?
```

Comment:
Also using this logic seems to be a bit much, you can assume that the new group has no leader.

Reply:
So the leader post for new group should not be created with `CreatePost.call`?

Reply:
I think this change is unnecessary since the `CreateGroup` does not rely on this

Reply:
I think in the end this one could be removed too.